lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -169,11 +169,7 @@ def merge_json_log(record)
         log = record['log'].strip
         if log[0].eql?('{') && log[-1].eql?('}')
           begin
-            parsed_log = JSON.parse(log)
-            record     = record.merge(parsed_log)
-            unless parsed_log.has_key?('log')
-              record.delete('log')
-            end
+            record = JSON.parse(log).merge(record)
```

Comment:
merge overwrites duplicate keys. So here I think the right thing to do is
merge the original record over top of the parsed one. Besides keeping 'log'
it retains the stream and timestamp entries.

On Wed, Sep 30, 2015 at 2:45 PM, Jimmi Dyson <notifications@github.com>
wrote:

> In lib/fluent/plugin/filter_kubernetes_metadata.rb
> <https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/pull/7#discussion_r40834151>
> :
>
> > @@ -169,11 +169,7 @@ def merge_json_log(record)
> >          log = record['log'].strip
> >          if log[0].eql?('{') && log[-1].eql?('}')
> >            begin
> > -            parsed_log = JSON.parse(log)
> > -            record     = record.merge(parsed_log)
> > -            unless parsed_log.has_key?('log')
> > -              record.delete('log')
> > -            end
> > +            record = JSON.parse(log).merge(record)
>
> This flips the merge logic effectively from record.merge(log) to
> log.merge(record). Which way round do we want to do this? Can't remember
> what the semantics for this are - only matters for conflicting keys I guess.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/pull/7/files#r40834151>
> .
>